refactor: Remove usages of reactable from TimeTable

kgabryje opened the pull request:
### SUMMARY
The goal is to replace all uses of `reactable-arc` library with DataTable from superset-ui (which uses `react-table` under the hood) and keep the UI/UX mostly unchanged. The reason for that is that `reactable-arc` has been unsupported for 2 years now, uses deprecated lifecycle methods and is incompatible with Typescript.
This PR replaces `reactable-arc` in `TimeTable` component.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

![image](https://user-images.githubusercontent.com/15073128/94174974-4a1a2b80-fe96-11ea-8f62-e3f06e37360a.png)

After:

![image](https://user-images.githubusercontent.com/15073128/94416605-585e9500-017f-11eb-8473-a83a97266ab1.png)

### TEST PLAN
<!--- What steps should be taken to verify the changes -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

kgabryje committed:
Use ListView instead of reactable

mistercrunch commented:
From the screenshot, wondering why the sparkline isn't expanding to 100% width (?)

rusackas reviewed (commented):
Code LGTM, but I'd like to get 👀 from @nytai on the ListView tweaks and implementation.

In regard to @mistercrunch's concern, the code looks basically the same, i.e. the sparkline has
`width={parseInt(column.width, 10) || 300}`. @kgabryje can we repro whether or not the code on `master has the same layout with these chart columns?

kgabryje committed:
Refactor TimeTable

kgabryje committed:
Spark column width fix

kgabryje commented:
@mistercrunch Thanks for pointing that out! A style prop for spark column wasn't passed correctly which caused it to have larger width than it should have. Fixed now. I updated the PR summary with the new screenshot.
![image](https://user-images.githubusercontent.com/15073128/94416545-45e45b80-017f-11eb-8b81-7229d0ddfb0b.png)
